server/services/teamsSummaryCountService.test.ts

```diff
@@ -0,0 +1,579 @@
+  })
+
+  it('should return products from object response', async () => {
+    const mockProducts = [{ id: 1, attributes: { name: 'Product 1', p_id: 'p1' } }]
+
+    strapiApiClient.getTeam.mockResolvedValue({
+      data: {
+        attributes: {
+          name: 'Team Name',
+          t_id: 'team-id',
+          products: {
+            data: mockProducts,
+          },
+          createdAt: '2023-01-01T00:00:00Z',
+          createdBy: { data: { attributes: {}, id: 1 } },
+          updatedAt: '2023-01-01T00:00:00Z',
+          updatedBy: { data: { attributes: {}, id: 1 } },
+        },
+      },
+    } as TeamResponse)
+
+    const result = await service.getProductsForTeam('team-slug')
+
+    expect(result).toEqual(mockProducts)
+  })
+
+  it('should return empty array when no products found', async () => {
+    strapiApiClient.getTeam.mockResolvedValue({
+      data: [
+        {
+          id: 1,
+          attributes: {
+            name: 'Team Name',
+            t_id: 'team-id',
+            products: {
+              data: [],
+            },
+            createdAt: '2023-01-01T00:00:00Z',
+            createdBy: { data: { attributes: {}, id: 1 } },
+            updatedAt: '2023-01-01T00:00:00Z',
+            updatedBy: { data: { attributes: {}, id: 1 } },
+          },
+        },
+      ],
+    } as TeamResponse)
+
+    const result = await service.getProductsForTeam('team-slug')
+
+    expect(result).toEqual([])
+    expect(mockLogger.info).toHaveBeenCalledWith(expect.stringContaining('Found 0 products for team team-slug'))
+  })
+
+  it('should return empty array on error', async () => {
+    strapiApiClient.getTeam.mockRejectedValue(new Error('API Error'))
+
+    const result = await service.getProductsForTeam('team-slug')
+
+    expect(result).toEqual([])
+    expect(mockLogger.error).toHaveBeenCalledWith(
+      expect.stringContaining('Error fetching products for team team-slug'),
+      expect.any(Error),
+    )
+  })
+})
+
+describe('TeamsSummaryCountService.getComponentsForProducts', () => {
+  let service: TeamsSummaryCountService
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    service = new TeamsSummaryCountService(mockAlertsService, () => mockStrapiClient)
+  })
+
+  it('should fetch components for multiple products', async () => {
+    const mockProducts = [
+      { attributes: { name: 'Product 1', slug: 'product-1', p_id: 'p1' } },
+      { attributes: { name: 'Product 2', slug: 'product-2', p_id: 'p2' } },
+    ] as ProductListResponseDataItem[]
+
+    const mockComponents1: ComponentListResponseDataItem[] = [
+      { id: 1, attributes: { name: 'Comp1A' } },
+      { id: 2, attributes: { name: 'Comp1B' } },
+    ]
+
+    const mockComponents2: ComponentListResponseDataItem[] = [{ id: 3, attributes: { name: 'Comp2A' } }]
+
+    strapiApiClient.getProduct.mockImplementation((params: ProductParams) => {
```

Comment:
Not happy with these. You're rewriting the implementation of functions you're using. Shouldn't need to change the implementation, just change the return value.